### PR TITLE
Dynamically generate ref tests

### PR DESCRIPTION
**Pull request by sfackler**
This uses the rustc-test crate, a copy of the standard test crate, to
dynamically create tests for each reference test. No need to remember to
update the macro, just add the directory to ref!

It also removes the line_wrap test, as it was never actually enabled and doesn't pass :( cc @mbrumlow

### CODE REVIEW

File: tests/ref.rs [hunk 1]
```diff
@@ -1,99 +1,78 @@
...
-        tmux_htop,
-        tmux_git_log,
-        vim_large_window_scroll,
-        indexed_256_colors,
-        fish_cc
+    let mut terminal = Term::new(size);
+    let mut parser = ansi::Processor::new();
+
+    for byte in recording {
+        parser.advance(&mut terminal, byte, &mut io::sink());
```
Comment (jwilm) [inline tests/ref.rs]: TIL about `io::Sink` 😄

**Comment by jwilm** [comment]
It seems that this changes the default logging behavior to not show individual test names. Is there an easy way to get the old behavior back? I tried configuring the options manually in the ref test setup, but the inline tests (ie in mods under src/) still were quiet.

**Comment by sfackler** [comment]
That change was made here: https://github.com/SimonSapin/rustc-test/commit/e3a9adca0d4592a5266046d57bd4b9bfa973e1c6. Not sure if there's a great way to get it back easily in the inline tests unfortunately :(

**Comment by jwilm** [comment]
This feature is still a big win so I'm going to merge it. I enabled the "capture" feature so all the stdout noise would go away.

Thanks again for putting this together!

**Pull request merged**